pkg/streamingpromql/operators/functions/range_vectors.go

```diff
@@ -533,3 +533,61 @@ func linearRegression(head, tail []promql.FPoint, interceptTime int64) (slope, i
 	intercept = sumY/n - slope*sumX/n
 	return slope, intercept
 }
+
+var Irate = FunctionOverRangeVectorDefinition{
+	SeriesMetadataFunction: DropSeriesName,
+	StepFunc:               irate(true),
+}
+
+var Idelta = FunctionOverRangeVectorDefinition{
+	SeriesMetadataFunction: DropSeriesName,
+	StepFunc:               irate(false),
+}
+
+func irate(isRate bool) RangeVectorStepFunction {
+	return func(step *types.RangeVectorStepData, _ float64, _ types.EmitAnnotationFunc) (float64, bool, *histogram.FloatHistogram, error) {
+		fHead, fTail := step.Floats.UnsafePoints()
+
+		// We need at least two samples to calculate irate or idelta.
+		if len(fHead)+len(fTail) < 2 {
```

Comment:
(nit) `len(fTail)` (and head) are used a lot below. Consider moving to a variable.

Reply:
Addressed in 5383c9cc0dcf10fd046b00314d929a42f753b263

Reply:
Suggest to move the lenTail/lenHead to above this line so you can use it here too

Reply:
Ahh you are right. Moved up in 1b1f3591afcf6d12df817ae972a4fb5e0cc4e6c9